mysql-test/suite/galera/t/galera_concurrent_ctas.test

```diff
@@ -1,56 +1,53 @@
 --source include/galera_cluster.inc
 --source include/big_test.inc
+--source include/have_debug_sync.inc
+
+#
+# To have real concurrent CTAS this test uses DEBUG_SYNC
+#
+
+--connect node_1a, 127.0.0.1, root, , test, $NODE_MYPORT_1
+
+--connection node_1
+#
+# Send CTAS it will block before open_tables call
+#
+SET DEBUG_SYNC = 'wsrep_create_table_as_select WAIT_FOR continue';
+--send CREATE table t1 as SELECT SLEEP(0.1);
+
+--connection node_1a
+#
+# Wait for CTAS to block
+#
+--echo # Wait until CTAS is on debug sync point
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.PROCESSLIST WHERE STATE = 'debug sync point: wsrep_create_table_as_select'
+--source include/wait_condition.inc
+
+--connection node_2
+#
+# Send concurrent CTAS
+#
+--send CREATE table t1 as SELECT SLEEP(0.2);
+
+--connection node_1a
+--echo # Wait until CTAS has executed
+SET DEBUG_SYNC= 'now SIGNAL continue';
+--let $wait_condition = SELECT COUNT(*) = 0 FROM INFORMATION_SCHEMA.PROCESSLIST WHERE STATE = 'Creating table	CREATE table t1 as SELECT SLEEP(0.1)'
+--source include/wait_condition.inc
+SET DEBUG_SYNC= 'RESET';
```

Comment:
this wait may not always succeed, there is nothing stopping the first CREATE TABLE after it was resumed from sync point. Therefore the processlist state change further to something else

Reply:
I wait here until CTAS disappears from process list i.e. COUN(*) = 0. Idea is that CTAS should here continue before sync point is reseted.